near-fetch/src/lib.rs

```diff
@@ -253,18 +269,24 @@ impl From<Client> for JsonRpcClient {
     }
 }
 
-async fn fetch_tx_errs(result: &FinalExecutionOutcomeView) -> Vec<&TxExecutionError> {
+async fn fetch_tx_errs(result: &RpcTransactionResponse) -> Vec<TxExecutionError> {
     let mut failures = Vec::new();
-
-    if let FinalExecutionStatus::Failure(tx_err) = &result.status {
-        failures.push(tx_err);
+    let outcome = result
+        .final_execution_outcome
+        .as_ref()
+        .unwrap()
+        .clone() // Clone to consume
+        .into_outcome();
+
+    if let FinalExecutionStatus::Failure(tx_err) = &outcome.status {
+        failures.push(tx_err.clone());
     }
-    if let ExecutionStatusView::Failure(tx_err) = &result.transaction_outcome.outcome.status {
-        failures.push(tx_err);
+    if let ExecutionStatusView::Failure(tx_err) = &outcome.transaction_outcome.outcome.status {
+        failures.push(tx_err.clone());
     }
-    for receipt in &result.receipts_outcome {
+    for receipt in &outcome.receipts_outcome {
         if let ExecutionStatusView::Failure(tx_err) = &receipt.outcome.status {
-            failures.push(tx_err);
+            failures.push(tx_err.clone());
         }
     }
     failures
```

Comment:
Is there a cleaner way to do this? Ran into some weirdness with references and borrowing. Also I changed the function signature which might be a no-no

Reply:
One thing is that the option needs to be explicitly handled, and we don't need to clone here to go into an `into_outcome`. We can directly reference the members of the enum themselves and get the outcome as a reference. And with this, we do not need to change the function signature at all nor need to clone the errors at all:

```rs
async fn fetch_tx_errs(result: &RpcTransactionResponse) -> Vec<&TxExecutionError> {
    let mut failures = Vec::new();
    let Some(outcome) = result.final_execution_outcome.as_ref() else {
        return failures;
    };
    let outcome = match outcome {
        FinalExecutionOutcomeViewEnum::FinalExecutionOutcome(outcome) => outcome,
        FinalExecutionOutcomeViewEnum::FinalExecutionOutcomeWithReceipt(outcome) => {
            &outcome.final_outcome
        }
    };

    if let FinalExecutionStatus::Failure(tx_err) = &outcome.status {
        failures.push(tx_err);
    }
    if let ExecutionStatusView::Failure(tx_err) = &outcome.transaction_outcome.outcome.status {
        failures.push(tx_err);
    }
    for receipt in &outcome.receipts_outcome {
        if let ExecutionStatusView::Failure(tx_err) = &receipt.outcome.status {
            failures.push(tx_err);
        }
    }
    failures
}
```